src/main/java/com/esri/samples/displayinformation/sketch_on_map/SketchOnMapController.java

```diff
@@ -0,0 +1,315 @@
+    sketchEditor.start(SketchCreationMode.POLYLINE);
+  }
+
+  /**
+   * Clear selection of any graphic in graphics overlay and start new polygon sketch
+   */
+  @FXML
+  private void handlePolygonButtonClicked() {
+    graphicsOverlay.clearSelection();
+    sketchEditor.start(SketchCreationMode.POLYGON);
+  }
+
+  /**
+   * Clear selection of any graphic in graphics overlay and start new freehand polyline sketch
+   */
+  @FXML
+  private void handleFreehandPolylineButtonClicked() {
+    graphicsOverlay.clearSelection();
+    sketchEditor.start(SketchCreationMode.FREEHAND_LINE);
+  }
+
+  /**
+   * Clear selection of any graphic in graphics overlay and start new freehand polygon sketch
+   */
+  @FXML
+  private void handleFreehandPolygonButtonClicked() {
+    graphicsOverlay.clearSelection();
+    sketchEditor.start(SketchCreationMode.FREEHAND_POLYGON);
+  }
+
+  /**
+   * Undo the last change made while sketching is active
+   */
+  @FXML
+  private void handleUndoButtonClicked() {
+    if (sketchEditor.canUndo()) {
+      sketchEditor.undo();
+    }
+  }
+
+  /**
+   * Redo the last change made while sketching is active
+   */
+  @FXML
+  private void handleRedoButtonClicked() {
+    if (sketchEditor.canRedo()) {
+      sketchEditor.redo();
+    }
+  }
+
+  /**
+   * Save the sketched graphic to the graphics overlay, and set its symbol to the type relevant for the geometry.
+   */
+  @FXML
+  private void handleSaveButtonClicked() {
+    // if the sketch isn't valid, stop the sketch editor.
+    if (!sketchEditor.isSketchValid()) {
+      sketchEditor.stop();
+      return;
+    }
+
+    Geometry sketchGeometry = sketchEditor.getGeometry();
+
+    // if an existed graphic being edited: get the selected graphic, set its geometry to that of the sketch editor geometry
+    // if a new graphic: create a new graphic based on the sketch editor geometry and set symbol depending on geometry type
+    if (sketchGeometry != null) {
+      if (!graphicsOverlay.getSelectedGraphics().isEmpty()) {
+        graphic = graphicsOverlay.getSelectedGraphics().get(0);
+        graphic.setGeometry(sketchGeometry);
+      } else {
+        graphic = new Graphic(sketchGeometry);
+
+        switch (sketchGeometry.getGeometryType()) {
+          case POLYGON:
+            graphic.setSymbol(fillSymbol);
+            break;
+          case POLYLINE:
+            graphic.setSymbol(lineSymbol);
+            break;
+          case POINT:
+          case MULTIPOINT:
+            graphic.setSymbol(pointSymbol);
+        }
+        graphicsOverlay.getGraphics().add(graphic);
+      }
+    }
+    sketchEditor.stop();
+
+    // allow the user to select a graphic from the map view
+    selectGraphic();
+    graphicsOverlay.clearSelection();
+    disableButtons();
+
+    if (!graphicsOverlay.getGraphics().isEmpty()) {
+      clearButton.setDisable(false);
+    }
+    stopButton.setDisable(true);
+  }
+
+  /**
+   * clear the graphics overlay of any saved graphics
```

Comment:
Javadoc is a little more strict than inline comments. Should capitalize first letter and have proper punctuation.